src/app/view/components/side-navigation-menu/side-navigation-menu.component.ts

```diff
@@ -74,7 +79,38 @@ export class SideNavigationMenuComponent implements AfterViewInit, OnDestroy {
     }
   }
 
-  constructor(private elementRef: ElementRef) {}
+  constructor(
+    private elementRef: ElementRef,
+    private getAllMenuSystemMenu: GetAllMenuSystemMenuUsecase
+  ) {}
+  ngOnInit(): void {
+    this._items = navigation.map((item) => {
+      item.path = this.checkPath(item.path);
+      return { ...item, expanded: !this._compactMode };
+    });
+    this.getAllMenuSystemMenu.execute({}).subscribe((grid: PageResultModel<SystemMenuModel>) => {
+      var mappedItems = grid.data?.map((item) => this.itemMenuMap(item));
+      if (mappedItems) {
+        this._mappedItems = mappedItems;
+      }
+    });
+  }
+
+  checkPath(path: string | undefined) {
+    return path && !/^\//.test(path) ? `/${path}` : path;
+  }
+
+  itemMenuMap(item: SystemMenuModel) {
+    var items = item.submenus?.map((item) => this.itemMenuMap(item));
```

Comment:
```suggestion
    var items = item.submenus?.map(this.itemMenuMap);
```
novamente pode utilizar desta forma.

Reply:
* Tanto para esse caso quanto para o anterior.

A função itemMenuMap acessa o "this" na sua implementação em dois pontos:
```TS
    itemMenuMap(item: SystemMenuModel) {
    var items = item.submenus?.map((item) => this.itemMenuMap(item));
    const mappedItem: MappedItem = {
      path: this.checkPath(item.route),  // USA O "this"
      text: item.title,
      icon: item.icon,
      expanded: !this._compactMode, // USA O "this"
      items: items,
    };
    return mappedItem;
  }
```
e dentro de arrow functions o this não existe, e isso acaba gerando um bug: https://github.com/anirudh-modi/JS-essentials/blob/master/ES2015/Functions/Arrow%20functions.md#how-this-is-different-for-arrow-functions